download/download.go

```diff
@@ -112,11 +114,19 @@ func (d *Downloader) doStart(context.Context) {
 	done := <-d.stop // blocks until there's something to read
 	cancel()
 	d.wg.Wait()
+	d.stopped = true
 	close(done)
 }
 
 // Stop tells the Downloader to stop downloading bundles.
 func (d *Downloader) Stop(context.Context) {
+	d.mtx.Lock()
+	defer d.mtx.Unlock()
+
+	if d.stopped {
+		return
```

Comment:
So, we basically want to make `Stop` idempotent here, right? So it can be called any number of times and it'll only do its thing once.

I suppose there're other ways to do that -- [sends to a closed channel block forever](https://dave.cheney.net/2014/03/19/channel-axioms), maybe we could use that in some way...but thinking about it for a hot minute, I don't think it'll be more obvious than what we do here. 🤔

Reply:
>So, we basically want to make Stop idempotent here, right? So it can be called any number of times and it'll only do its thing once.

Yes, that's the goal.